ext/sodium/config.m4

```diff
@@ -1,5 +1,5 @@
 dnl $Id$
-dnl config.m4 for extension libsodium
+dnl config.m4 for extension sodium
 
 PHP_ARG_WITH(libsodium, for libsodium support,
 [  --with-libsodium[[=DIR]]  Include libsodium support])
```

Comment:
I think we need 2 options
- one for the ext (--enable-sodium)
- one optional to set the library directory (--with-libsodium=DIR)

And for consistency all "libsodium" fonction (minit, minfo, ...) should also be renamed